migrations/lib/importer/steps/badge_groupings.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Migrations::Importer::Steps
+  class BadgeGroupings < ::Migrations::Importer::CopyStep
+    MAX_NAME_LENGTH = 100
+    MAX_DESCRIPTION_LENGTH = 500
+
+    store_mapped_ids true
+
+    requires_mapping :ids_by_name, "SELECT LOWER(name), id FROM badge_groupings"
+
+    column_names %i[id name description position created_at updated_at]
+
+    total_rows_query <<~SQL, MappingType::BADGE_GROUPINGS
+      SELECT COUNT(*)
+      FROM badge_groupings
+           LEFT JOIN mapped.ids mapped_badge_grouping
+             ON badge_groupings.original_id = mapped_badge_grouping.original_id
+               AND mapped_badge_grouping.type = ?
+      WHERE mapped_badge_grouping.original_id IS NULL
+    SQL
+
+    rows_query <<~SQL, MappingType::BADGE_GROUPINGS
+      SELECT badge_groupings.*,
+             ROW_NUMBER() OVER (
+              ORDER BY COALESCE(badge_groupings.position, 0), badge_groupings.original_id
```

Comment:
What are your thoughts on ordering by `ROWID` instead? If there's no `position` set, then the insertion order of the converter might be a better clue than the `original_id`.


```suggestion
              ORDER BY COALESCE(badge_groupings.position, 0), badge_groupings.ROWID
```